jobspec/parse_test.go

```diff
@@ -1865,3 +1865,26 @@ func TestIncorrectKey(t *testing.T) {
 		t.Fatalf("Expected key error; got %v", err)
 	}
 }
+
+// TestPortParsing validates that the removal of the mapstructure tags on the Port
+// struct don't cause issues with HCL 1 parsing.
```

Comment:
This comment will be stale and mysterious after we ship this. Probably worth having a TODO here that points to a new issue examining whether we should have `mapstructure` tags on _any_ of the API at this point.

Reply:
I added a comment that clarifies this.